### ISSUE TITLE
Date format has invisible characters in IE11

### ISSUE THREAD

**Issue by MattHo** [formatjs/react-intl#201]
formatDate function from react-intl returns date with U+200E (left-to-right marker) characters in IE11. Since the same function returns date without these characters in server, this causes checksum mismatch. Please fix formatDate() to return date string without any invisible characters.

**Comment by ericf** [comment]
There is a detailed description and discussion of this issue at the ECMA 402 spec level: https://github.com/tc39/ecma402/issues/28

I searched through all the CLDR locale data used by the `Intl.js` polyfill and `\u200E` isn't used for dates, but it _is used_ for Farsi (`fa`) currency formatting, it would be rendered incorrectly if stripped from `fa` when formatting numbers.

We also can't blinding strip the LRMs and RLMs characters they are there for a reason — to provide bidi hits on structured text. If they are removed it will lead to incorrect and incomprehensible strings in the UI.

```js
var df = new Intl.DateTimeFormat('ar', {
    day  : 'numeric', 
    month: 'numeric', 
    year : 'numeric'
});

var withRLMs    = df.format(Date.now());
var withoutRLMs = df.format(Date.now()).replace(/\u200F/g, '');

console.log(withRLMs);    // "٣‏/١١‏/٢٠١٥"
console.log(withoutRLMs); // "٣/١١‏/٢٠١٥"
```

In the above example, the expected output for Arabic is "year/month/day", if the RLMs are removed, it ends up being "year/day/month" which is incorrect and would confuse the user.

The issue here is that IE adds some redundant, but valid, LRM chars, these are safe to strip in most cases, **but not** the Farsi one mentioned above.

----

There's a larger issue here for dates and React checksum though: **timezones**. While the ECMA 402 spec and JavaScript implementations support timezones for date/time formatting, the `Intl.js` polyfill does not (because of the size of the data), and the polyfill is required for Node 0.10. Node 0.12+ has built-in `Intl` support and therefore supports specifying a timezone.

Even if IE used the CLDR like Chrome and Firefox for its backing locale data, we still have the issue of Node 0.10 and timezones being different from server to client causing checksums to fail. Because of this, I'm thinking of solution that also updates the client after `componentDidMount`.

With relative times, I've implemented a similar solution to avoid checksum mismatches: https://github.com/yahoo/react-intl/pull/191
